capstone/src/main/webapp/scripts/poll.js

```diff
@@ -1,37 +1,116 @@
 const BAR_WIDTH = "690";
 const BAR_HEIGHT = "55";
+const TRANSITION_MILLIS = 400;
+let maxVotes;
 
 /**
- * Get poll options from server and load to DOM.
+ * Add new option to poll.
+ */
+function addPollOption() {
+  const text = document.getElementById("input-box").value;
+  if (text.trim() === "") return;
+  document.getElementById("input-box").value = "";
+  fetch(`poll?text=${text}`, { method: "POST" }).then(
+    setTimeout(getPollOptions, 500)
+  );
+}
+
+/**
+ * Get poll data, which includes each poll option and the * list of options
+ * that the current logged in user has voted for, from server and load to DOM.
  */
 function getPollOptions() {
   fetch("/poll")
     .then(response => response.json())
-    .then(options => {
+    .then(pollData => {
       const optionsContainer = document.getElementById("options-container");
       optionsContainer.innerHTML = "";
-      options.forEach(option => {
-        renderOptionElement(option);
+      let maxVotes = getMaxVotes(pollData);
+      pollData["options"].forEach(option => {
+        renderOptionElement(option, maxVotes);
       });
+      return pollData["votedOptions"];
+    })
+    .then(votedOptions => {
+      handleCheck(votedOptions);
     });
 }
 
+/**
+ * Gets the maximum number of votes for the poll. The option with maximum
+ * number of votes should be the first option since the options are sorted in
+ * order of votes.
+ * @param {objct} pollData
+ */
+function getMaxVotes(pollData) {
+  let maxVotes;
+  // Gets the list of votes for the first poll option
+  let votesArray = pollData["options"][0]["votes"];
+  if (votesArray == null) {
+    maxVotes = 0;
+  } else {
+    maxVotes = votesArray.length;
+  }
+  return maxVotes;
```

Comment:
I think you can make this more succinct with a ternary operator:
 ```suggestion
  // Gets the list of votes for the first poll option
  let votesArray = pollData["options"][0]["votes"];
  return !!votesArray ? votesArray.length : 0;
```
In case you don't know the "!!" returns the boolean truthy/falsy association of a value: https://developer.mozilla.org/en-US/docs/Glossary/Truthy
Here it'll equate to true unless the array is null